projects/SDE Lighthouse 03 - Kontopantelis/template-sql/4.outcomes.template.sql

```diff
@@ -0,0 +1,127 @@
+--┌───────────────────────────────────┐
+--│ Outcomes for dementia cohort      │
+--└───────────────────────────────────┘
+
+---- RESEARCH DATA ENGINEER CHECK ----
+
+--------------------------------------
+
+-- From application:
+--  Table 4: Outcomes (2006 to present)
+--  - PatientID
+--  - OutcomeName (e.g. dementia care review, referral to social prescribing)
+--  - OutcomeDate
+
+--  Outcomes include: 
+--  - Care processes
+--      - dementia care reviews (INCLUDED HERE)
+--      - anticholinergic medication burden (IN MEDICATION FILE)
+--      - potentially inappropriate prescribing rates (IN MEDICATION FILE)
+--      - antipsychotic use (IN MEDICATION FILE)
+--      - medication review within 6 weeks of commencing an antipsychotic (INCLUDED HERE PLUS MEDICATION FILE)
+--      - medication review (INCLUDED HERE)
+--      - appropriate anti-dementia medication prescribing (IN MEDICATION FILE)
+--      - advance care planning (NOT INCLUDED - WAS OPTIONAL FROM PI)
+--      - continuity of care measures (NOT INCLUDED - WAS OPTIONAL FROM PI - ALSO WE DON'T HAVE CLINICIAN LEVEL DATA SO NOT POSSIBLE)
+--      - carer type (NOT INCLUDED - WAS OPTIONAL FROM PI)
+--      - carer review (NOT INCLUDED - WAS OPTIONAL FROM PI)
+--      - referrals to social prescribing (INCLUDED HERE)
+--      - social care referrals (INCLUDED HERE)
+--      - safeguarding referrals (INCLUDED HERE)
+--  - Healthcare utilisation factors:
+--      - frequency of attendance  (INCLUDED HERE - gp encounter and hospital admission by type)
+--      - missed appointments (NOT INCLUDED - WAS OPTIONAL FROM PI)
+--  - Key adverse clinical outcomes: 
+--      - All-cause mortality (IN THE PATIENTS FILE)
+--      - unscheduled hospital admissions (INCLUDED HERE)
+--      - delirium (INCLUDED HERE)
+--      - falls (INCLUDED HERE)
+--      - fractures (INCLUDED HERE)
+
+
+--TODO|> CODESET advance-care-planning:1
+--> CODESET delirium:1 fracture:1 falls:1 social-care-referral:1 safeguarding-referral:1
+
+
+DROP TABLE IF EXISTS OutcomeCodes;
+CREATE TEMPORARY TABLE OutcomeCodes AS
+SELECT GmPseudo, "SuppliedCode", to_date("EventDate") as EventDate
+FROM {{cohort-table}} cohort
+LEFT OUTER JOIN INTERMEDIATE.GP_RECORD."GP_Events_SecondaryUses" events 
+    ON events."FK_Patient_ID" = cohort.FK_Patient_ID
+WHERE "SuppliedCode" IN (SELECT code FROM {{code-set-table}} WHERE concept IN (
+	'delirium','fracture','falls','social-care-referral','advance-care-planning','safeguarding-referral'
+));
+
+DROP TABLE IF EXISTS {{project-schema}}."4_Outcomes";
+CREATE TABLE {{project-schema}}."4_Outcomes" AS
+-- gp admissions
+select "GmPseudo" AS "PatientID", 'GP encounter' AS "OutcomeName", "EventDate" AS "OutcomeDate"
+from "Contacts_Proxy"
+WHERE "GmPseudo" IN (SELECT GmPseudo FROM {{cohort-table}})
+AND "EventDate" >= '2006-01-01'
+UNION
+-- hospital admissions
+select top 100
+    SUBSTRING("Der_Pseudo_NHS_Number", 2)::INT AS "GmPseudo",
+    CASE
```

Comment:
That's useful to know thanks. In this case I think I'll leave it because I think the default is too granular for the project and some of the descriptions might be confusing. So given I'd need to group them anyway I might as well do it from the admission method codes.